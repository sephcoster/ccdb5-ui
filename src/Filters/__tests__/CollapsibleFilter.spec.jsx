import React from 'react';
import { mount } from 'enzyme';
import CollapsibleFilter from '../CollapsibleFilter';
import renderer from 'react-test-renderer';

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = renderer.create(
      <CollapsibleFilter title="foo" desc="bar" />
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('component:CollapsibleFilter', () => {
  it('hides the children when Hide is clicked', () => {
    const target = mount(<CollapsibleFilter showChildren={true} />);
    const theButton = target.find('.o-expandable_cue button')

    expect(target.state('showChildren')).toEqual(true);
    theButton.simulate('click');
    expect(target.state('showChildren')).toEqual(false);
  })    
})
