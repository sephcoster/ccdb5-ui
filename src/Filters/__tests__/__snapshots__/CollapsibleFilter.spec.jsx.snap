// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state renders without crashing 1`] = `
<section
  className="o-expandable"
>
  <div
    className="o-expandable_header o-expandable_target"
  >
    <h4
      className="o-expandable_header-left o-expandable_label"
    >
      foo
    </h4>
    <span
      className="o-expandable_header-right o-expandable_link"
    >
      <button
        className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
        onClick={[Function]}
      >
        Hide
        <span
          className="cf-icon cf-icon-minus-round"
        />
      </button>
    </span>
  </div>
  <p>
    bar
  </p>
</section>
`;
